File: Solved/2000/2622/2622.py
import math
N = int(input())
first = int(math.ceil(N / 3))
second = N // 2 - (0 if N & 1 else 1)

answer = 0
for C in range(first, second + 1):
    answer += (C - (N - C) // 2)
    if not (N - C) & 1:
        answer += 1
print(answer)

File: Solved/2000/2622/2622.cpp
#include <iostream>
using namespace std;
int main(){
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
    
    int N; cin >> N;
    int answer = 0;
    for(int first = 1; first <= N / 3; first++){
        for(int second = (N - first) / 2; second >= first; second--){
            if (N - first - second < first + second){
                answer++;
            }
            else{
                break;
            }
        }
    }
    cout << answer;
}
